gslib: add shared library support

cameronrutherford opened the pull request:
This has been working for local development of palace, and I would like feedback on these changes. I don't think these changes will break anything, but perhaps shared and static builds can be enabled at the same time.

cameronrutherford committed:
gslib: add shared library support

cameronrutherford commented:
@spackbot style

cameronrutherford committed:
Fix spack style.

alalazo commented on var/spack/repos/builtin/packages/gslib/package.py:
```diff
@@ -39,7 +40,10 @@ class Gslib(Package):
         src_dir = "src"
         lib_dir = "lib"
-        libname = "libgs.a"
+        if "+shared" in spec:
+            libname = "libgs.so"
+        else:
+            libname = "libgs.a"
```
(Nitpick) This can be moved below, near the place where it's used.

alalazo commented on var/spack/repos/builtin/packages/gslib/package.py:
(Optional, not related to the changes in the PR) At the end of the recipe there are a few nested conditionals that can be simplified. Also we might want to deprecate a few old versions (v1.0.0, v1.0.1), so we can cleanup the recipe for the release in November.

alalazo reviewed (dismissed):
Additions look good to me. I commented with a few optional changes to improve the recipe. If you want to implement them later let me know and I'll merge this.

cameronrutherford commented on var/spack/repos/builtin/packages/gslib/package.py:
```diff
@@ -48,6 +52,12 @@ def install(self, spec, prefix):
 
         cc = self.compiler.cc
```
@alalazo my IDE is saying "Cannot access attribute "cc" for class "DeprecatedCompiler"... I think this might reflect that no-one builds this library without MPI?

Or at least this needs to be fixed as me move to compilers as nodes.

becker33 replied:
Yeah I think it needs to change to `self["c"].cc`

cameronrutherford replied:
```suggestion
        cc = self["c"].cc
```

Like this? I think we can apply the suggestion and then merge...

cameronrutherford committed:
Address PR review.

cameronrutherford commented:
Thanks for the review @alalazo. I updated and addressed your comments, although I am still confused about the compiler as nodes changes that I commented on

cameronrutherford committed:
gslib: update c compiler spec
